src/hl/plist.rs

```diff
@@ -202,6 +203,20 @@ impl PropertyList {
     }
 }
 
+pub fn set_vlen_manager(plist: crate::internal_prelude::hid_t) -> Result<()> {
```

Comment:
```suggestion
pub fn set_vlen_manager(plist: hid_t) -> Result<()> {
```

Reply:
> Maybe add `// TODO: move this to dataset_transfer module when DatasetTransfer plist is implemented`? (so it doesn't hang out here in the wild)
> 
> Also, I don't think this name is clear enough; two alternative options:
> 
>     1. `pub fn set_vlen_manager_libc(plist: hid_t)`
> 
>     2. `pub fn set_vlen_manager<A: Fn(usize) -> *mut u8, F: Fn(*mut u8)>(plist: hid_t, alloc: A, free: F)`
> 
> 
> In (2), you'd call it as
> 
> ```rust
> set_vlen_manager(plist_id, |size| unsafe { libc::malloc(size) }, |ptr| unsafe { libc::free(ptr) })
> ```
> 
> (assuming the catch_unwind stuff can remain in the generic function itself)
> 
> You could probably even have both, where (1) calls (2).

(2) requires quite a bit of trickery, as we have to get a function pointer from a closure. The easiest way would be to get a closure pointer, wrap this up in the `alloc_info` and `free_info`, and call this from a trampoline. This is a bit overkill when we are tied to the same allocator as for the varlen types in `h5-types` .